src/lib/analytics.js

```diff
@@ -138,25 +138,64 @@ store.subscribe(({isSignedIn}) => {
   ga('set', dimensions.SIGNED_IN, isSignedIn ? '1' : '0');
 });
 
-/**
- * Add a listener to detect back/forward cache restores and track them
- * as pageviews with the "bfcache" navigation type set (in case we need
- * to distinguish them from regular pageviews).
- * https://web.dev/bfcache/#how-bfcache-affects-analytics-and-performance-measurement
- */
-window.addEventListener(
-  'pageshow',
+function sendPageView() {
   /**
-   * @param {PageTransitionEvent} e
+   * Add a listener to detect back/forward cache restores and track them
+   * as pageviews with the "back-forward-cache" navigation type set (in
+   * case we need to distinguish them from regular pageviews).
+   * https://web.dev/bfcache/#how-bfcache-affects-analytics-and-performance-measurement
    */
-  (e) => {
-    if (e.persisted) {
-      ga('set', dimensions.NAVIGATION_TYPE, 'back-forward-cache');
-      ga('send', 'pageview');
-    }
-  },
-);
+  window.addEventListener(
+    'pageshow',
+    /**
+     * @param {PageTransitionEvent} e
+     */
+    (e) => {
+      if (e.persisted) {
+        ga('set', dimensions.NAVIGATION_TYPE, 'back-forward-cache');
+        ga('send', 'pageview');
+      }
+    },
+  );
+
+  // Handle prerenders that have already activated
+  if (
+    performance &&
+    performance.getEntriesByType &&
+    performance.getEntriesByType('navigation')[0].activationStart > 0
+  ) {
+    ga('set', dimensions.NAVIGATION_TYPE, 'prerender');
+    ga('send', 'pageview');
+    return;
+  }
+
+  // Handle prerenders that activate in the future
+  if (document.prerendering) {
```

Comment:
I think it probably makes sense to defer everything in this file until after the page activates, and I think that'll make the code much easier to follow as well.

Perhaps something like this:

```js
const whenActivated = new Promise((resolve) => {
  if (document.prerendering) {
    document.addEventListener('prerenderingchange', () => resolve());
  } else {
    resolve();
  }
});
```

...and then you can create an async `initAnalytics` function that awaits that promise as the first thing it does and then sends the pageview and initializes everything else afterward:

```js
async function initAnalytics() {
  await whenActivated;

  // 1) Set all custom dimensions...

  // 2) Then send the pageview 

  // 3) Set up pageshow listener for bfcache pageviews...

  // 4) Register web vitals functions...
}

initAnalytics();
```